File: backend/src/main/java/com/team01/backend/domain/comment/dto/CommentResponseDto.java
package com.team01.backend.domain.comment.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.team01.backend.domain.comment.entity.Comment;

import java.time.LocalDateTime;

public record CommentResponseDto(
        Long id,
        String content,
        String author,
        int likeCount,
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
        LocalDateTime createdAt,
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
        LocalDateTime modifiedAt
) {
    // Comment 엔티티를 받아서 record 만드는 정적 팩토리 메서드
    public static CommentResponseDto from(Comment comment) {
        return new CommentResponseDto(
                comment.getId(),
                comment.getContent(),
                comment.getUser().getNickname(),
                comment.getLikeCount(),
                comment.getCreatedAt(),
                comment.getModifiedAt()
        );
    }
}

File: backend/src/main/java/com/team01/backend/domain/comment/dto/CommentResponseDto.kt
package com.team01.backend.domain.comment.dto

import com.fasterxml.jackson.annotation.JsonFormat
import com.team01.backend.domain.comment.entity.Comment
import java.time.LocalDateTime

data class CommentResponseDto(
    val id: Long,
    val content: String,
    val author: String,
    val likeCount: Int,
    @field:JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    val createdAt: LocalDateTime,
    @field:JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    val modifiedAt: LocalDateTime,
) {
    companion object {
        // 컨벤션: 정적 팩토리 메서드 of 사용
        fun of(comment: Comment): CommentResponseDto =
            CommentResponseDto(
                id = comment.id,
                content = comment.content,
                author = comment.user.nickname,
                likeCount = comment.likeCount,
                createdAt = comment.createdAt,
                modifiedAt = comment.modifiedAt,
            )

        fun from(comment: Comment): CommentResponseDto = of(comment)
    }
}
